projects/batfish/src/test/java/org/batfish/job/BatfishJobExecutorTest.java

```diff
@@ -0,0 +1,218 @@
+    executor.initializeJobsStats(Lists.newArrayList(new BfTestJob(settings, "result1")));
+
+    // Simulating finishing of a job and handling the result
+    BfTestResult bfTestResult = new BfTestResult(1000L, _logger.getHistory(), "result");
+    Set<String> output = new HashSet<>();
+    List<BatfishException> failureCauses = new ArrayList<>();
+    BfTestAnswerElement ae = new BfTestAnswerElement();
+    executor.updateJobsStats();
+    executor.handleJobResult(bfTestResult, output, ae, failureCauses);
+
+    // checking the log of the executor for the job finished
+    assertEquals(
+        _logger.getHistory().toString(400),
+        String.format(
+            "Job terminated successfully with result: %s after elapsed time: %s - %d/%d "
+                + "(%.1f%%) complete\n",
+            bfTestResult.toString(),
+            CommonUtil.getTime(bfTestResult.getElapsedTime()),
+            1,
+            1,
+            100.0));
+  }
+
+  @Test
+  public void testHandleJobResultFailure() {
+    Settings settings = new Settings();
+    settings.setHaltOnParseError(false);
+
+    // initializing executor
+    BatfishJobExecutor<BfTestJob, BfTestAnswerElement, BfTestResult, Set<String>> executor =
+        new BatfishJobExecutor<>(
+            settings, _logger, settings.getHaltOnParseError(), "Test Job Executor");
+    executor.initializeJobsStats(Lists.newArrayList(new BfTestJob(settings, "result1")));
+
+    // Simulating failure of a job and handling the result
+    BfTestResult bfTestResult =
+        new BfTestResult(
+            1000L, _logger.getHistory(), new BatfishException("Test Job Failure Message"));
+    Set<String> output = new HashSet<>();
+    List<BatfishException> failureCauses = new ArrayList<>();
+    BfTestAnswerElement ae = new BfTestAnswerElement();
+    executor.updateJobsStats();
+    executor.handleJobResult(bfTestResult, output, ae, failureCauses);
+
+    //checking that correct failure message is written in the log
+    assertEquals(
+        failureCauses.get(0).getMessage(),
+        String.format(
+            "Failure running job after elapsed time: %s\n-----"
+                + "BEGIN JOB LOG-----\n\n-----END JOB LOG-----",
+            CommonUtil.getTime(bfTestResult.getElapsedTime())));
+  }
+
+  @Test
+  public void testHandleProcessingError() {
+    Settings settings = new Settings();
+    settings.setHaltOnParseError(true);
+
+    // initializing executor
+    BatfishJobExecutor<BfTestJob, BfTestAnswerElement, BfTestResult, Set<String>> executor =
+        new BatfishJobExecutor<>(
+            settings, _logger, settings.getHaltOnParseError(), "Test Job Executor");
+    List<BatfishException> failureCauses = new ArrayList<>();
+
+    // checking if the exception thrown has correct class and message
+    _thrown.expect(CompositeBatfishException.class);
+    _thrown.expectMessage("Fatal exception due to failure of at least one job");
```

Comment:
In general I'd prefer that you do not expect messages from `Exception` classes that we control. Instead, subclass the `Exception` and simply expect the appropriate type. If there is really only one usage of the exception in question, then no test of the message is necessary.

Reply:
done